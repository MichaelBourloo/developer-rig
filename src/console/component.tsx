import * as React from 'react';
import './component.sass';

interface ConsoleEntry {
  frame: string;
  log: string;
}

interface State {
  logHistory: ConsoleEntry[];
}

export class Console extends React.Component<{}, State> {
  public state: State = {
    logHistory: window.rig.history || []
  }

  updateConsole() {
    this.setState({
      logHistory: window.rig.history
    });
  }

  componentDidMount() {
    window.rig.update = () => {
      this.updateConsole();
    }

    this.updateConsole();
  }

  render() {
    const logs = this.state.logHistory.map((entry, index) => (
      <div key={index}>{entry.frame} $ {entry.log}</div>
    ));

    return (
      <div className="console">{logs}</div>
    );
  }
}
