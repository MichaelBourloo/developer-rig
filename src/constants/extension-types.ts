const { ExtensionAnchor, ExtensionPlatform } = window['extension-coordinator'];

export const DEFAULT_EXTENSION_TYPE = ExtensionAnchor.Overlay;
export const ExtensionAnchors: { [key: string]: string } = {
  [ExtensionAnchor.Overlay]: 'Overlay',
  [ExtensionAnchor.Panel]: 'Panel',
  [ExtensionAnchor.Component]: 'Component',
  [ExtensionPlatform.Mobile]: 'Mobile',
};

