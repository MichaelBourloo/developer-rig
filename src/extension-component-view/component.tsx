import * as React from 'react';
import { ExtensionFrame } from '../extension-frame';
import { RigExtension, ViewStyles, FrameSize } from '../core/models/rig';
import { Position } from '../types/extension-coordinator';
const { ExtensionMode, ExtensionViewType, getComponentPositionFromView, getComponentSizeFromView } = window['extension-coordinator'];

export interface ExtensionComponentViewProps {
  id: string
  extension: RigExtension;
  frameSize: FrameSize;
  position: Position;
  role: string;
}

type Props = ExtensionComponentViewProps & React.HTMLAttributes<HTMLDivElement>;

export class ExtensionComponentView extends React.Component<Props> {
  private computeViewStyles(): ViewStyles {
    const extension = this.props.extension;
    const positionFromView = getComponentPositionFromView(
      this.props.frameSize.width,
      this.props.frameSize.height,
      {
        x: this.props.position.x * 100,
        y: this.props.position.y * 100,
      });
    const sizeFromView = getComponentSizeFromView(
      this.props.frameSize.width,
      this.props.frameSize.height,
      extension.views.component);

    let viewStyles: ViewStyles = {
      border: '1px solid #7D55C7',
      position: 'absolute',
      left: positionFromView.x + 'px',
      top: positionFromView.y + 'px',
      width: `${sizeFromView.width}px`,
      height: `${sizeFromView.height}px`,
    }

    if (extension.views.component.zoom) {
      viewStyles = {
        ...viewStyles,
        width: `${sizeFromView.width / sizeFromView.zoomScale}px`,
        height: `${sizeFromView.height / sizeFromView.zoomScale}px`,
        transformOrigin: '0 0',
        transform: `scale(${sizeFromView.zoomScale})`,
      }
    }

    return viewStyles;
  }

  public render() {
    return (
      <div
        className="view component-view"
        style={{
          width: this.props.frameSize.width + 'px',
          height: this.props.frameSize.height + 'px',
        }}>
          <div style={this.computeViewStyles()}>
          <ExtensionFrame
            className="view"
            frameId={`frameid-${this.props.id}`}
            extension={this.props.extension}
            type={ExtensionViewType.Component}
            mode={ExtensionMode.Viewer}
          />
        </div>
      </div>
    );
  }
}
