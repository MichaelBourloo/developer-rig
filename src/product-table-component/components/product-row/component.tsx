import * as React from 'react';
import { Product, ValidationErrors } from '../../../core/models/product';

export interface ProductRowProps {
  product: Product;
  handleValueChange: any;
  handleDeprecateClick: any;
}

export interface State {
  validationErrors: ValidationErrors;
  invalid: boolean;
}
export class ProductRow extends React.Component<ProductRowProps, State>{
  constructor(props: ProductRowProps) {
    super(props);
    this.state = {
      validationErrors: props.product.validationErrors || {},
      invalid: Object.keys(props.product.validationErrors).length > 0,
    }
  }

  public render() {
    return (
      <div className="product-table__row">
        <div className="text-col">
          <input type="text"
            className={this.state.validationErrors.displayName ? "form__input invalid" : "form__input"}
            name="displayName"
            required
            maxLength={255}
            value={this.props.product.displayName}
            onChange={this.props.handleValueChange}
          />
          {this.state.invalid &&
            <p className="invalid-hint">
              {this.state.validationErrors.displayName}
            </p>
          }
        </div>
        <div className="text-col">
          <input type="text"
            className={this.state.validationErrors.sku ? "form__input invalid" : "form__input"}
            name="sku"
            required
            maxLength={255}
            disabled={this.props.product.savedInCatalog}
            value={this.props.product.sku}
            onChange={this.props.handleValueChange}
          />
          {this.state.invalid &&
            <p className="invalid-hint">
              {this.state.validationErrors.sku}
            </p>
          }
        </div>
        <div className="text-col">
          <input type="number"
            className={this.state.validationErrors.amount ? "form__input invalid" : "form__input"}
            name="amount"
            required
            min="1" max="10000"
            value={this.props.product.amount}
            onChange={this.props.handleValueChange}
          />
          {this.state.invalid &&
            <p className="invalid-hint">
              {this.state.validationErrors.amount}
            </p>
          }
        </div>
        <div className="select-col">
          <select name="inDevelopment"
              className={this.state.validationErrors.inDevelopment ? "form__input invalid" : "form__input"}
              value={this.props.product.inDevelopment}
              onChange={this.props.handleValueChange}>
            <option value="true">Yes</option>
            <option value="false">No</option>
          </select>
          {this.state.invalid &&
            <p className="invalid-hint">
              {this.state.validationErrors.inDevelopment}
            </p>
          }
        </div>
        <div className="select-col">
          <select name="broadcast"
              className={this.state.validationErrors.broadcast ? "form__input invalid" : "form__input"}
              value={this.props.product.broadcast}
              onChange={this.props.handleValueChange}>
            <option value="true">Yes</option>
            <option value="false">No</option>
          </select>
          {this.state.invalid &&
            <p className="invalid-hint">
              {this.state.validationErrors.broadcast}
            </p>
          }
        </div>
        <div className="button-col">
          <button className="product-row__deprecate-button"
              onClick={this.props.handleDeprecateClick}>
            {this.props.product.deprecated ? 'Restore' : 'Deprecate'}
          </button>
          {this.state.invalid && <p className="invalid-hint"></p>}
        </div>
        <div className="dirty-col">
        <div className={this.props.product.dirty ? "dirty-indicator" : "dirty-indicator hidden"}>
          <svg xmlns='http://www.w3.org/2000/svg' width='30' height='30'>
            <circle cx='5' cy='5' r='3' />
          </svg>
        </div>
        {this.state.invalid && <p className="invalid-hint"></p>}
        </div>
      </div>
    );

  }
}
