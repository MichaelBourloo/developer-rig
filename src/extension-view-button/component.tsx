import * as React from 'react';
import * as line from '../img/rig_line.png';
import * as plus from '../img/plus_icon.png';
import './component.sass'

export interface ExtensionViewButtonProps {
  onClick: () => void;
}

type Props = ExtensionViewButtonProps;
export class ExtensionViewButton extends React.Component<Props> {
  render() {
    return (
      <div className='button-wrapper'>
        <img alt='line' className='background' src={line} />
        <div className="circle-button" onClick={this.props.onClick}>
          <img alt='Add new view' src={plus} />
        </div>
      </div>
    )
  }
}
