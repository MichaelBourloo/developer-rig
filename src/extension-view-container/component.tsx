import * as React from 'react';
import { ExtensionView } from '../extension-view';
import { ExtensionViewButton } from '../extension-view-button';
import { EXTENSION_MODE_TO_VIEW } from '../constants/extension-modes'
import './component.sass';
import { RigExtensionView } from '../core/models/rig';
const { ExtensionMode } = window['extension-coordinator'];

export interface ExtensionViewContainerProps {
  mode: string;
  extensionViews: RigExtensionView[];
  openEditViewHandler?: Function;
  deleteExtensionViewHandler: Function;
  openExtensionViewHandler: Function;
  extension: object;
}

export class ExtensionViewContainer extends React.Component<ExtensionViewContainerProps> {
  openExtensionViewDialog = () => {
    this.props.openExtensionViewHandler();
  }

  render() {
    if (this.props.mode !== ExtensionMode.Viewer) {
      return ( <ExtensionView
          id={this.props.mode}
          type={EXTENSION_MODE_TO_VIEW[this.props.mode]}
          extension={this.props.extension}
          mode={this.props.mode}
          key={this.props.mode}/>
      );
    }

    let extensionViews: RigExtensionView[] = [];
    if (this.props.extensionViews && this.props.extensionViews.length > 0) {
      extensionViews = this.props.extensionViews.map(view => {
        return <ExtensionView
          key={view.id}
          id={view.id}
          extension={view.extension}
          type={view.type}
          mode={this.props.mode}
          role={view.role}
          frameSize={view.frameSize}
          position={{x: view.x, y: view.y}}
          linked={view.linked}
          orientation={view.orientation}
          openEditViewHandler={this.props.openEditViewHandler}
          deleteViewHandler={this.props.deleteExtensionViewHandler}/>
      });
    }

    return (
      <div className='view-container-wrapper'>
          <div className="view-container">
          { extensionViews }
        </div>
        <div>
          <ExtensionViewButton
            onClick={this.openExtensionViewDialog}>
          </ExtensionViewButton>
        </div>
      </div>
    );
  }
}
