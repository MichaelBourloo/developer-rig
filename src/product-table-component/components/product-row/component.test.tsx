import { setupShallowTest } from '../../../tests/enzyme-util/shallow';
import { ProductRow } from './component';
import * as TestData from '../../../tests/constants/products';

describe('<ProductRow />', () => {
  const setupShallow = setupShallowTest(ProductRow, () => ({
    product: TestData.TestProduct1
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('shows validation errors', () => {
    const { wrapper } = setupShallow({
      product: {
        ...TestData.TestProduct1,
        validationErrors: {
          sku: 'SKU is invalid'
        }
      }
    });
    expect(wrapper.find('input.invalid[name="sku"]')).toHaveLength(1);
    expect(wrapper.find('p.invalid-hint').filterWhere(n => n.text() === 'SKU is invalid')).toHaveLength(1);
  });

  it('shows row as dirty', () => {
    const { wrapper } = setupShallow({
      product: TestData.UnsavedProduct,
    });
    expect(wrapper.find('.dirty-indicator')).toHaveLength(1);
  });
});
