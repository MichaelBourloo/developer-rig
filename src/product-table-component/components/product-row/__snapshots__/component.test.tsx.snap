// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductRow /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductRow
    product={
        Object {
            "amount": 100,
            "broadcast": "false",
            "deprecated": false,
            "dirty": false,
            "displayName": "Test Product 1",
            "inDevelopment": "false",
            "savedInCatalog": true,
            "sku": "testSku1",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="text-col"
>
          <input
                    className="form__input"
                    maxLength={255}
                    name="displayName"
                    onChange={undefined}
                    required={true}
                    type="text"
                    value="Test Product 1"
          />
</div>,
        <div
          className="text-col"
>
          <input
                    className="form__input"
                    disabled={true}
                    maxLength={255}
                    name="sku"
                    onChange={undefined}
                    required={true}
                    type="text"
                    value="testSku1"
          />
</div>,
        <div
          className="text-col"
>
          <input
                    className="form__input"
                    max="10000"
                    min="1"
                    name="amount"
                    onChange={undefined}
                    required={true}
                    type="number"
                    value={100}
          />
</div>,
        <div
          className="select-col"
>
          <select
                    className="form__input"
                    name="inDevelopment"
                    onChange={undefined}
                    value="false"
          >
                    <option
                              value="true"
                    >
                              Yes
                    </option>
                    <option
                              value="false"
                    >
                              No
                    </option>
          </select>
</div>,
        <div
          className="select-col"
>
          <select
                    className="form__input"
                    name="broadcast"
                    onChange={undefined}
                    value="false"
          >
                    <option
                              value="true"
                    >
                              Yes
                    </option>
                    <option
                              value="false"
                    >
                              No
                    </option>
          </select>
</div>,
        <div
          className="button-col"
>
          <button
                    className="product-row__deprecate-button"
                    onClick={undefined}
          >
                    Deprecate
          </button>
</div>,
        <div
          className="dirty-col"
>
          <div
                    className="dirty-indicator hidden"
          >
                    <svg
                              height="30"
                              width="30"
                              xmlns="http://www.w3.org/2000/svg"
                    >
                              <circle
                                        cx="5"
                                        cy="5"
                                        r="3"
                              />
                    </svg>
          </div>
</div>,
      ],
      "className": "product-table__row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form__input"
              maxLength={255}
              name="displayName"
              onChange={undefined}
              required={true}
              type="text"
              value="Test Product 1"
/>,
            undefined,
          ],
          "className": "text-col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form__input",
              "maxLength": 255,
              "name": "displayName",
              "onChange": undefined,
              "required": true,
              "type": "text",
              "value": "Test Product 1",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form__input"
              disabled={true}
              maxLength={255}
              name="sku"
              onChange={undefined}
              required={true}
              type="text"
              value="testSku1"
/>,
            undefined,
          ],
          "className": "text-col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form__input",
              "disabled": true,
              "maxLength": 255,
              "name": "sku",
              "onChange": undefined,
              "required": true,
              "type": "text",
              "value": "testSku1",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form__input"
              max="10000"
              min="1"
              name="amount"
              onChange={undefined}
              required={true}
              type="number"
              value={100}
/>,
            undefined,
          ],
          "className": "text-col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form__input",
              "max": "10000",
              "min": "1",
              "name": "amount",
              "onChange": undefined,
              "required": true,
              "type": "number",
              "value": 100,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="form__input"
              name="inDevelopment"
              onChange={undefined}
              value="false"
>
              <option
                            value="true"
              >
                            Yes
              </option>
              <option
                            value="false"
              >
                            No
              </option>
</select>,
            undefined,
          ],
          "className": "select-col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="true"
>
                  Yes
</option>,
                <option
                  value="false"
>
                  No
</option>,
              ],
              "className": "form__input",
              "name": "inDevelopment",
              "onChange": undefined,
              "value": "false",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Yes",
                  "value": "true",
                },
                "ref": null,
                "rendered": "Yes",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No",
                  "value": "false",
                },
                "ref": null,
                "rendered": "No",
                "type": "option",
              },
            ],
            "type": "select",
          },
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="form__input"
              name="broadcast"
              onChange={undefined}
              value="false"
>
              <option
                            value="true"
              >
                            Yes
              </option>
              <option
                            value="false"
              >
                            No
              </option>
</select>,
            undefined,
          ],
          "className": "select-col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="true"
>
                  Yes
</option>,
                <option
                  value="false"
>
                  No
</option>,
              ],
              "className": "form__input",
              "name": "broadcast",
              "onChange": undefined,
              "value": "false",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Yes",
                  "value": "true",
                },
                "ref": null,
                "rendered": "Yes",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No",
                  "value": "false",
                },
                "ref": null,
                "rendered": "No",
                "type": "option",
              },
            ],
            "type": "select",
          },
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="product-row__deprecate-button"
              onClick={undefined}
>
              Deprecate
</button>,
            undefined,
          ],
          "className": "button-col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Deprecate",
              "className": "product-row__deprecate-button",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Deprecate",
            "type": "button",
          },
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="dirty-indicator hidden"
>
              <svg
                            height="30"
                            width="30"
                            xmlns="http://www.w3.org/2000/svg"
              >
                            <circle
                                          cx="5"
                                          cy="5"
                                          r="3"
                            />
              </svg>
</div>,
            undefined,
          ],
          "className": "dirty-col",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <svg
                height="30"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
>
                <circle
                                cx="5"
                                cy="5"
                                r="3"
                />
</svg>,
              "className": "dirty-indicator hidden",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <circle
                  cx="5"
                  cy="5"
                  r="3"
/>,
                "height": "30",
                "width": "30",
                "xmlns": "http://www.w3.org/2000/svg",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "cx": "5",
                  "cy": "5",
                  "r": "3",
                },
                "ref": null,
                "rendered": null,
                "type": "circle",
              },
              "type": "svg",
            },
            "type": "div",
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="text-col"
>
            <input
                        className="form__input"
                        maxLength={255}
                        name="displayName"
                        onChange={undefined}
                        required={true}
                        type="text"
                        value="Test Product 1"
            />
</div>,
          <div
            className="text-col"
>
            <input
                        className="form__input"
                        disabled={true}
                        maxLength={255}
                        name="sku"
                        onChange={undefined}
                        required={true}
                        type="text"
                        value="testSku1"
            />
</div>,
          <div
            className="text-col"
>
            <input
                        className="form__input"
                        max="10000"
                        min="1"
                        name="amount"
                        onChange={undefined}
                        required={true}
                        type="number"
                        value={100}
            />
</div>,
          <div
            className="select-col"
>
            <select
                        className="form__input"
                        name="inDevelopment"
                        onChange={undefined}
                        value="false"
            >
                        <option
                                    value="true"
                        >
                                    Yes
                        </option>
                        <option
                                    value="false"
                        >
                                    No
                        </option>
            </select>
</div>,
          <div
            className="select-col"
>
            <select
                        className="form__input"
                        name="broadcast"
                        onChange={undefined}
                        value="false"
            >
                        <option
                                    value="true"
                        >
                                    Yes
                        </option>
                        <option
                                    value="false"
                        >
                                    No
                        </option>
            </select>
</div>,
          <div
            className="button-col"
>
            <button
                        className="product-row__deprecate-button"
                        onClick={undefined}
            >
                        Deprecate
            </button>
</div>,
          <div
            className="dirty-col"
>
            <div
                        className="dirty-indicator hidden"
            >
                        <svg
                                    height="30"
                                    width="30"
                                    xmlns="http://www.w3.org/2000/svg"
                        >
                                    <circle
                                                cx="5"
                                                cy="5"
                                                r="3"
                                    />
                        </svg>
            </div>
</div>,
        ],
        "className": "product-table__row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form__input"
                maxLength={255}
                name="displayName"
                onChange={undefined}
                required={true}
                type="text"
                value="Test Product 1"
/>,
              undefined,
            ],
            "className": "text-col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "maxLength": 255,
                "name": "displayName",
                "onChange": undefined,
                "required": true,
                "type": "text",
                "value": "Test Product 1",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form__input"
                disabled={true}
                maxLength={255}
                name="sku"
                onChange={undefined}
                required={true}
                type="text"
                value="testSku1"
/>,
              undefined,
            ],
            "className": "text-col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "disabled": true,
                "maxLength": 255,
                "name": "sku",
                "onChange": undefined,
                "required": true,
                "type": "text",
                "value": "testSku1",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form__input"
                max="10000"
                min="1"
                name="amount"
                onChange={undefined}
                required={true}
                type="number"
                value={100}
/>,
              undefined,
            ],
            "className": "text-col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "max": "10000",
                "min": "1",
                "name": "amount",
                "onChange": undefined,
                "required": true,
                "type": "number",
                "value": 100,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                className="form__input"
                name="inDevelopment"
                onChange={undefined}
                value="false"
>
                <option
                                value="true"
                >
                                Yes
                </option>
                <option
                                value="false"
                >
                                No
                </option>
</select>,
              undefined,
            ],
            "className": "select-col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="true"
>
                    Yes
</option>,
                  <option
                    value="false"
>
                    No
</option>,
                ],
                "className": "form__input",
                "name": "inDevelopment",
                "onChange": undefined,
                "value": "false",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yes",
                    "value": "true",
                  },
                  "ref": null,
                  "rendered": "Yes",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No",
                    "value": "false",
                  },
                  "ref": null,
                  "rendered": "No",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                className="form__input"
                name="broadcast"
                onChange={undefined}
                value="false"
>
                <option
                                value="true"
                >
                                Yes
                </option>
                <option
                                value="false"
                >
                                No
                </option>
</select>,
              undefined,
            ],
            "className": "select-col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="true"
>
                    Yes
</option>,
                  <option
                    value="false"
>
                    No
</option>,
                ],
                "className": "form__input",
                "name": "broadcast",
                "onChange": undefined,
                "value": "false",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yes",
                    "value": "true",
                  },
                  "ref": null,
                  "rendered": "Yes",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No",
                    "value": "false",
                  },
                  "ref": null,
                  "rendered": "No",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="product-row__deprecate-button"
                onClick={undefined}
>
                Deprecate
</button>,
              undefined,
            ],
            "className": "button-col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Deprecate",
                "className": "product-row__deprecate-button",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Deprecate",
              "type": "button",
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="dirty-indicator hidden"
>
                <svg
                                height="30"
                                width="30"
                                xmlns="http://www.w3.org/2000/svg"
                >
                                <circle
                                                cx="5"
                                                cy="5"
                                                r="3"
                                />
                </svg>
</div>,
              undefined,
            ],
            "className": "dirty-col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <svg
                  height="30"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
>
                  <circle
                                    cx="5"
                                    cy="5"
                                    r="3"
                  />
</svg>,
                "className": "dirty-indicator hidden",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <circle
                    cx="5"
                    cy="5"
                    r="3"
/>,
                  "height": "30",
                  "width": "30",
                  "xmlns": "http://www.w3.org/2000/svg",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "cx": "5",
                    "cy": "5",
                    "r": "3",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "circle",
                },
                "type": "svg",
              },
              "type": "div",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
